### ISSUE TITLE
Iterable type variable parameter becomes unknown in function returning iterator

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1206]
**Describe the bug**
Annotating a function to return `Iterator[T]` for an `Iterator` subclass which implements a constructor accepting any `Iterable[T]` produces an inscrutable error that the type of the `Iterator`'s parameter is unknown:

```
  13:12 - error: Return type, "foo[Unknown]", is partially unknown (reportUnknownVariableType)
  18:12 - error: Return type, "chain[Unknown]", is partially unknown (reportUnknownVariableType)
```

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That lines 13 and 18 would not produce an error.

**Screenshots or Code**
```python
# pyright: strict

from typing import Iterable, Iterator, TypeVar

T = TypeVar('T')

class foo(Iterator[T]):
    def __init__(self, it: Iterable[T]) -> None: ...
    def __next__(self) -> T: ...
    def __iter__(self) -> Iterator[T]: ...

def bar(it: Iterable[T]) -> Iterator[T]:
    return foo(it)

def baz(it_of_its: Iterable[Iterable[T]]) -> Iterator[T]:
    from itertools import chain

    return chain(*it_of_its)

```

**VS Code extension or command-line**
Either, 1.1.88

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.89, which I just published. It will also appear in the next Pylance release.
